htdocs/product/stock/product.php

```diff
@@ -643,17 +649,26 @@
 			$textdesc = $langs->trans("CostPriceDescription");
 			$textdesc .= "<br>".$langs->trans("CostPriceUsage");
 			$text = $form->textwithpicto($langs->trans("CostPrice"), $textdesc, 1, 'help', '');
-			print $form->editfieldkey($text, 'cost_price', $object->cost_price, $object, $usercancreate, 'amount:6');
-			print '</td><td>';
-			print $form->editfieldval($text, 'cost_price', $object->cost_price, $object, $usercancreate, 'amount:6');
-			print '</td></tr>';
+			if (!$usercancreadprice) {
+				print $form->editfieldkey($text, 'cost_price', '', $object, 0, 'amount:6');
+				print '</td><td>';
+				print $form->editfieldval($text, 'cost_price', '', $object, 0, 'amount:6');
+				print '</td></tr>';
```

Comment:
Because the opening of tr is outside the if (!$usercanreadprice), the the closing of tr mus also be outside the if.

Reply:
oups sorry it's done